docs/architecture/maui/communicating-between-components.md

```diff
@@ -1,115 +1,115 @@
+
+The eShopOnContainers multi-platform app uses the `WeakReferenceMessenger` class to communicate between loosely coupled components. The app defines a single message named `AddProductMessage`. The `AddProductMessage` is published by the `CatalogViewModel` class when an item is added to the shopping basket. In return, the `CatalogView` class subscribes to the message and uses this to highlight the product adds with animation in response.
 
-In the eShopOnContainers multi-platform app, `MessagingCenter` is used to update in the UI in response to an action occurring in another class. Therefore, messages are published on the UI thread, with subscribers receiving the message on the same thread.
+In the eShopOnContainers multi-platform app, `WeakReferenceMessenger` is used to update the UI in response to an action occurring in another class. Therefore, messages are published from the thread that the class is executing on, with subscribers receiving the message on the same thread.
 
 > [!TIP]
 > Marshal to the UI or main thread when performing UI updates. If updates to user interfaces are not made on this thread, it can cause the application to crash or become unstable.
 
 If a message that's sent from a background thread is required to update the UI, process the message on the UI thread in the subscriber by invoking the `MainThread.BeginInvokeOnMainThread` method.
 
-For more information about `MessagingCenter`, see [MessagingCenter](/dotnet/maui/fundamentals/messagingcenter) on the Microsoft Developer Center.
+For more information about `Messenger`, see [Messenger](/dotnet/communitytoolkit/mvvm/messenger) on the Microsoft Developer Center.
 
 ## Defining a message
 
-`MessagingCenter` messages are strings that are used to identify messages. The following code example shows the messages defined within the eShopOnContainers multi-platform app:
+`IMessenger` messages are custom objects that provide custom payloads. The following code example shows the `AddProductMessage` message defined within the eShopOnContainers multi-platform app:
 
 ```csharp
-public static class MessengerKeys
+public class AddProductMessage : ValueChangedMessage<int>
 {
-    // Add product to basket
-    public const string AddProduct = nameof(AddProduct);
+    public AddProductMessage(int count) : base(count)
+    {
+    }
 }
 ```
 
-In this example, messages are defined using constants. The advantage of this approach is that it provides compile-time type safety and refactoring support.
+The base class is defined using `ValueChangedMessage<T>` where T can be of any type needed to pass data. Both message publishers and subscribers can expect messages of a specific type (i.e. `AddProductMessage`). This can help ensure that both parties have agreed to a messaging contract and that the data provided with that contract will be consistent. Additionally, this approach provides compile-time type safety and refactoring support.
```

Comment:
```suggestion
The base class is defined using `ValueChangedMessage<T>` where `T` can be of any type needed to pass data. Both message publishers and subscribers can expect messages of a specific type (for example, `AddProductMessage`). This can help ensure that both parties have agreed to a messaging contract and that the data provided with that contract will be consistent. Additionally, this approach provides compile-time type safety and refactoring support.
```

Per style guide, avoid things like `i.e.` or `e.g.`.